globalflags.go

```diff
@@ -38,18 +45,84 @@ func (gf *globalFlags) ensureDockerDaemon() {
 }
 
 func requireRepo(conf config, prefs schemaPrefs, fl *flag.FlagSet) repo {
-	repoURI := fl.Arg(0)
+	var (
+		repoURI = strings.Join(fl.Args(), "/")
+		r       repo
+		err     error
+	)
+
 	if repoURI == "" {
 		flog.Fatal("Argument <repo> must be provided.")
 	}
 
-	r, err := parseRepo(defaultSchema(conf, prefs), conf.DefaultHost, repoURI)
+	// if this returns a non-empty string know it's pointing to a valid project on disk
+	// an error indicates an existing path outside of the project dir
+	repoName, err := pathIsRunnable(conf, repoURI)
 	if err != nil {
-		flog.Fatal("failed to parse repo %q: %v", repoURI, err)
+		flog.Fatal(err.Error())
+	}
+
+	if repoName != "" {
+		// we only need the path since the repo exists on disk.
+		// there's not currently way for us to figure out the host anyways
+		r = repo{URL: &url.URL{Path: repoName}}
+	} else {
+		r, err = parseRepo(defaultSchema(conf, prefs), conf.DefaultHost, repoURI)
+		if err != nil {
+			flog.Fatal("failed to parse repo %q: %v", repoURI, err)
+		}
+	}
+
+	// check if path is pointing to a subdirectory
+	if sp := strings.Split(r.Path, "/"); len(sp) > 2 {
+		r.Path = strings.Join(sp[:2], "/")
+		r.subdir = strings.Join(sp[2:], "/")
 	}
+
 	return r
 }
 
+// pathIsRunnable returns the container name if the given path exists and is
+// in the projects directory, else an empty string An error is returned if
```

Comment:
```suggestion
// in the projects directory. Otherwise an empty string and an error is returned if
```

Reply:
It's possible for an empty string and a nil error to be returned if the path doesn't exist inside of the project. The error is only if the specified directory exist and is outside of the projects directory

Reply:
I think I misunderstood, imo the error is more of a separate thought

Reply:
Ok that sounds good, then I guess just a `.` is needed after `else an empty string`

Reply:
oh oops